<?xml version="1.0"?>
<launch>
    <arg name="rosmastername" default="SATELLITE-S50-B"/>
    <machine name="tsn_caffe" address="tsn_caffe" user="root" password="ros_ros" env-loader="/temporal-segment-networks/catkin_ws/src/caffe_tsn_ros/env.sh $(arg rosmastername)" ssh-port="22"/>
  	<arg name="camera_name" default="videofiles" />
  	<!-- set camera fps to -->
  	<arg name="set_camera_fps" default="30" />
  	<!-- set buffer queue size of frame capturing to -->
  	<arg name="buffer_queue_size" default="1000" />
  	<!-- frames per second to query the camera for -->
  	<arg name="fps" default="30" />
  	<!-- frame_id for the camera -->
  	<arg name="frame_id" default="$(arg camera_name)" />
  	<!-- By default, calibrations are stored to file://${ROS_HOME}/camera_info/${NAME}.yaml
  	To use your own fill this arg with the corresponding url, e.g.:
  	"file:///$(find your_camera_package)/config/your_camera.yaml" -->
    <arg name="camera_info_url" default="" />
  	<!-- flip the image horizontally (mirror it) -->
  	<arg name="flip_horizontal" default="false" />
  	<!-- flip the image vertically -->
  	<arg name="flip_vertical" default="false" />
    <!-- force width and height, 0 means no forcing -->
    <arg name="width" default="0"/>
    <arg name="height" default="0"/>
  	<!-- if show a image_view window subscribed to the generated stream -->
	  <arg name="visualize" default="true"/>
    <arg name="base_path" default="/media/frederico/167e9587-675e-4fd2-baa0-1400b6594a90/var/datasets/videos/" />


   	<!-- images will be published at /camera_name/image with the image transports plugins (e.g.: compressed) installed -->
   	<group ns="$(arg camera_name)">
	    <node pkg="datasetloader_ros" type="video_stream_multiple_files" name="$(arg camera_name)_stream" output="screen">
	    	<remap from="camera" to="image_raw" />
	    	<param name="camera_name" type="string" value="$(arg camera_name)" />
	        <param name="set_camera_fps" type="double" value="$(arg set_camera_fps)" />
	        <param name="buffer_queue_size" type="int" value="$(arg buffer_queue_size)" />
	        <param name="fps" type="double" value="$(arg fps)" />
	        <param name="frame_id" type="string" value="$(arg frame_id)" />
	        <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
	        <param name="flip_horizontal" type="bool" value="$(arg flip_horizontal)" />
	        <param name="flip_vertical" type="bool" value="$(arg flip_vertical)" />
	        <param name="width" type="int" value="$(arg width)" />
	        <param name="height" type="int" value="$(arg height)" />
        <param name="autoplay" type="bool" value="false"/>
	    </node>

	    <node if="$(arg visualize)" name="$(arg camera_name)_image_view" pkg="image_view" type="image_view">
	    	<remap from="image" to="image_raw" />
	    </node>

      <node pkg="datasetloader_ros" type="readpathsrv" name="readpathnode" output="screen">
          <param name="basepath" type="string" value="$(arg base_path)" />
          <param name="splitdir" value="/home/frederico/Documents/catkin_ws/src/datasetloader_ros/data/hmdb51_7030splits"/>
      </node>
	</group>
  <node name="class_tsn_rgb" machine="tsn_caffe" pkg="caffe_tsn_ros" type="class_tsn_rgb2.py" respawn="false" output="screen">
    <param name="video_topic" value="videofiles/image_raw" />
    <param name="device_id" value="0" />
    <param name="dataset" value="hmdb51" />
    <param name="split" value="1" />
    <param name="classification_frame_window" value="50" />
  </node>
  <node if="$(arg visualize)" name="class_image_view" pkg="image_view" type="image_view">
    <remap from="image" to="/class_overlay_image_raw" />
  </node>
</launch>
